전남대 FE 박진현 - 1단계 : 로그인 기능 구현하기

jinhyun71744 opened the pull request:
검증 로직을 구현하면서 리랜더링이 어떻게 일어나는지 신경쓰는게 조금 어려웠던 것 같습니다. 
특히 커스텀 훅은 좀 더 잘 리팩토링할 수 있을 것 같은데 방법을 찾기가 어렵네요

Login.tsx에 있는 요소들을 컴포넌트로 따로 분리해볼까 생각하기도 했는데 재사용할 것 같지 않아 그냥 두었습니다.

jinhyun71744 committed:
feat: email 검증 기능 구현

- hooks/useLoginForm 커스텀 훅 구현
- 기존 페이지 이동 로직 삭제
- 이메일 검증 기능 구현

jinhyun71744 committed:
feat: PW 검증 기능 구현

- useLoginForm 훅 수정
- pw 검증 기능 구현

jinhyun71744 committed:
docs: README.md 수정

jinhyun71744 committed:
style: Home 페이지 UI 수정

- 그리드 열 수정
- 필터 양쪽정렬

jinhyun71744 committed:
feat: ErrorText 컴포넌트 구현

- ErrorText 컴포넌트 구현
- 절대경로 추가
- Login의 Input css 수정

jinhyun71744 committed:
style: 로그인 버튼 비활성화 기능 추가

jinhyun71744 committed:
refactor: useLoginForm 리팩토링

Coreight98 commented on src/components/common/ErrorText.tsx:
```diff
@@ -0,0 +1,21 @@
+
+interface Props {
+  children: React.ReactNode;
+}
```
react에서 제공해주는 PropsWithChildren 타입을 활용하면 좋을거 같아요!

jinhyun71744 replied:
리뷰 반영해서 수정했습니다! 확인 부탁드립니다

Coreight98 commented on src/hooks/useLoginForm.ts:
```diff
@@ -0,0 +1,64 @@
+
+    setEmailError('');
+    return true;
+  };
```
현재 validateEmail 함수는 2가지 일을 하는거 같아요

유효성을 검증하는 로직
오류 메시지 상태를 업데이트하는 로직
단일 책임을 가지도록 2가지 일을 분리할 수 있을거 같아요!

``` ts
 type ValidateResult = {
    ok: true;
  } | {
    ok: false;
    reason: string;
  };
  
  function validateEmail(email: string): ValidateResult {
    if (!id) {
      return { ok: false, reason: 'ID를 입력해주세요' };
    }
    if (!isEmail(id)) {
      return { ok: false, reason: 'ID는 이메일 형식으로 입력해주세요.' };
    }
    return { ok: true };
  }

  function handleCheckEmail() {
    const result = validateEmail(email);
    setIdError(result.ok ? '' : result.reason);
    return result.ok;
  }
```

jinhyun71744 replied:
리뷰 반영해서 수정했습니다! 확인 부탁드립니다

Coreight98 reviewed (changes requested):
진현님 안녕하세요! 금주 리뷰어 차재명입니다!
전반적으로 코드를 잘 작성해주셨고, 요구사항도 잘 반영해주신거 같아요! 간단하게 코멘드 남겼는데, 확인 한번 부탁드릴게요! 😊🙌🏻

jinhyun71744 committed:
refactor:  PropsWithChildren 적용

jinhyun71744 committed:
refactor: validateEmail, validatePassword 기능 분리

- validate*Format 함수 구현
- validateEmail, validatePassword 기능 분리
- 제안하신 handleCheck* 를 validate*로 구현 (다른 컴포넌트는 수정하지 않기 위함)

Coreight98 reviewed (approved):
진현님 고생 많으셨습니다! 👍 
2단계 진행해주시면 좋을거 같아요!
